lib/rspec/support/reentrant_mutex.rb

```diff
@@ -28,8 +30,8 @@ def synchronize
     private
 
       def enter
-        @mutex.lock if @owner != Thread.current
-        @owner = Thread.current
+        @mutex.lock if @owner != Fiber.current
+        @owner = Fiber.current
```

Comment:
WDYT of
```ruby
def current
  defined?(Fiber.current) ? Fiber.current : Thread.current
end
```
and using it?

Reply:
It's broken.

Reply:
Can you please elaborate?

Reply:
Whenever some file would `require 'fiber'` the behavior and what's stored here would switch and it'd be a total mess.

Reply:
> some file would require 'fiber' the behavior and what's stored here would switch and it'd be a total mess

For sure. I hope they don't do it dynamically when there are existing fibers and reentrant mutexes in action. Otherwise - do you see any issue?

Ok, let's put it another way. I couldn't find any mention of `Fiber` in stdlib docs. [Core docs](https://ruby-doc.org/core-2.7.1/Fiber.html#method-c-current) claim that in order to user `current`/`transfer`/`alive?` one has to `require 'fiber'`.
Unlike it is with `require 'rubygems'` or `require 'date'`, `require 'set'` etc no constants are loaded, and it's not a problem if we load it from RSpec.

Reply:
Would this not break in a more typical threaded solution? E.g. JRuby

Reply:
>  I hope they don't do it dynamically when there are existing fibers and reentrant mutexes in action.

I don't think it's advisable to take that risk/assumption, and anyway AFAIK there is no need.